src/main/java/io/pinecone/clients/Pinecone.java

```diff
@@ -99,6 +104,301 @@ public IndexModel createServerlessIndex(String indexName, String metric, int dim
         return indexModel;
     }
 
+    // Minimal
+    public IndexModel createPodsIndex(String indexName, Integer dimension, String environment, String podType) {
+        if (indexName == null || indexName.isEmpty()) {
```

Comment:
There's some repetition in each create methods which can be extracted out by creating a validateAndBuildCreatePodsIndexRequest(). I'm sure there's a better name so I'll leave it up to you but a similar approach for validating and building query request in data plane can be found here: 
https://github.com/pinecone-io/pinecone-java-client/blob/main/src/main/java/io/pinecone/commons/IndexInterface.java#L128


Note that the CreateIndexRequestSpecPod podSpec = new CreateIndexRequestSpecPod() can be done before validations and after each parameter is validated, we can keep setting the value.

Similarly, once podSpec object is all created and values are set, we can approach:
CreateIndexRequest createIndexRequest = new CreateIndexRequest();
and start setting the values. 

If the required parameter is missing, we would like to throw an exception.
If the optional parameter is missing, we dont want to set the value. 


Finally, your overloaded methods would look like this:
public IndexModel createPodsIndex(String indexName, Integer dimension, String environment, String podType) {
    return validateAndBuildCreatePodsIndex(indexName, dimension, environment, null, podType, null, null, null, null, null);
}

Reply:
Another option is just validating the required parameters by creating a new method:
private void validateIndexCreation(String indexName, Integer dimension, String environment, String podType) {
    if (indexName == null || indexName.isEmpty()) {
        throw new PineconeValidationException("indexName cannot be null or empty");
    }
    if (dimension == null) {
        throw new PineconeValidationException("Dimension cannot be null");
    }
    if (dimension < 1) {
        throw new PineconeValidationException("Dimension must be greater than 0. See limits for more info: https://docs.pinecone.io/reference/limits");
    }
    if (environment == null || environment.isEmpty()) {
        throw new PineconeValidationException("Environment cannot be null or empty");
    }
    if (podType == null || podType.isEmpty()) {
        throw new PineconeValidationException("Pod type cannot be null or empty");
    }
}